### DISCUSSION TITLE
Taskbar Clock Customization doesn't work on 22621

### DISCUSSION THREAD

**Discussion by win98se** [ramensoftware/windhawk-mods#13] [Q&A] [answered]
As in title; Installed the mod, doesn't work.

Are there anything I need to do to provide help in investigating the issue?

**Comment by m417z** [comment, marked as answer]
I haven't tried it specifically on 22621, but I saw that it doesn't work on my Windows 11. I released Taskbar Clock Customization v1.0.1 with a fix, please try it and let me know whether it fixes it for you.

**Reply by win98se** [reply]
Unfortunately, v1.0.1 still doesn't work here on 22621 (to be exact, currently build 22621.105). How do I provide useful information about this?

_More information: 22621 is confirmed as the RTM build of Windows 11 Version 22H2._

**Reply by m417z** [reply]
Can you upload your explorerextensions.dll file? If there are small changes, I might be able to figure it out. If not, I'll probably have to install the new version and debug it.

**Reply by win98se** [reply]
Unlike 22000, 22621 seems to not have ExplorerExtensions.dll, it has FileExplorerExtensions.dll instead -
![image](https://user-images.githubusercontent.com/11819816/174335677-dc3ada6d-15a4-484f-b8b2-e0ea24aa29a1.png)

For comparison, my friend's computer running 22000 has ExplorerExtensions.dll -
![image](https://user-images.githubusercontent.com/11819816/174336754-e0f6b107-f8ca-4362-a3d6-35d1f31bdee7.png)

This is the FileExplorerExtensions.dll extracted from my system, zipped - [FileExplorerExtensions.zip](https://github.com/ramensoftware/windhawk-mods/files/8929223/FileExplorerExtensions.zip)

**Reply by m417z** [reply]
I installed Windows 11 version 22H2. The file that contains the relevant code now is `Taskbar.View.dll` which is a new library. There were also other minor changes. I just released v1.0.2 which should work, please try it.

**Reply by win98se** [reply]
It finally works! Thank you for your huge efforts.

**Comment by win98se** [comment]
@m417z I now have another issue with it (as of v1.0.2) - it doesn't work on every boot, I have to manually disable and re-enable it for it to work again.

**Reply by m417z** [reply]
You're right. In fact, looks like it never worked on Windows 11, even with v1.0 and build 22000. I'll look into fixing it soon.

**Reply by win98se** [reply]
Thank you in advance!

**Reply by m417z** [reply]
Should be fixed now in v1.0.3.

**Reply by win98se** [reply]
Confirm fixed. Thank you!

**Reply by BruceLee569** [reply]
@m417z 

Hello, this mod doesn't work in my system too, the version of Taskbar Clock Customization installed online stays at `1.0.1`. 
So I manul add `1.0.3` version, but is not working too. 

My windows version is `21H2 (OS Internal version 22000.739)` ;

In this [article](https://www.windowslatest.com/2022/04/17/microsoft-explains-why-it-wont-add-seconds-to-windows-11-taskbar-clock/), I learned unlike Windows 10, Windows 11 doesn't support the optional registry key to display seconds on the taskbar. In a Feedback Hub post, Microsoft quietly explained that it is no longer possible to edit the Registry file to enable a clock with seconds. The company has removed the feature entirely and one of the reasons could be performance issues.

Before I discovered windhawk, I couldn't find any software that can display the seconds. I think your modular idea of windhawk is great!

**Reply by m417z** [reply]
It should work with Windows 11 build 22000. Perhaps you tried it while symbols for explorer were still being downloaded. The mod uses symbols which are automatically downloaded, and there's no indication for it, something I want to improve in one of the next versions.

If you waited enough time after installation and it still didn't work, let's create a new discussion topic and I'll try to help you figure out what's wrong.

**Comment by crz4all** [comment]
I was still using a beta version.
After I updated, it's working.
